tests/cases/conformance/controlFlow/dependentDestructuredVariables.ts

```diff
@@ -467,3 +467,20 @@ const parameterReassignedContextualRest1: (...args: [1, 2] | [3, 4]) => void = (
     x; // 1 | 3
   }
 }
+
+// https://github.com/microsoft/TypeScript/issues/59652
+
+declare function mutuallyEnabledPair(): {
+    discriminator: true,
+    value: string,
+  } | {
+    discriminator: false,
+    value: null | undefined,
+  }
+  
+  
+  export const { discriminator, value: value59652 } = mutuallyEnabledPair()

```

Comment:
You should make this a new test or remove export; note now the other baseline for this file completely changed.